packages/ai-chat/src/browser/session-summary-variable-contribution.ts

```diff
@@ -0,0 +1,115 @@
+import { AIVariableCompletionContext, FrontendVariableContribution, FrontendVariableService } from '@theia/ai-core/lib/browser';
+import { MaybePromise, QuickInputService, QuickPickItem, QuickPickItemOrSeparator, QuickPickSeparator } from '@theia/core';
+import { ChatService, ChatSession } from '../common';
+import { codiconArray } from '@theia/core/lib/browser';
+import * as monaco from '@theia/monaco-editor-core';
+import { ChatSessionSummaryAgent } from '../common/chat-session-summary-agent';
+
+export const SESSION_SUMMARY_VARIABLE: AIVariable = {
+    id: 'session-summary',
+    description: 'Resolves to a summary of the session with the given ID.',
+    name: 'session-summary',
+    label: 'Session Summary',
+    iconClasses: codiconArray('clippy'),
+    isContextVariable: true,
+    args: [{ name: 'session-id', description: 'The ID of the session to summarize.' }]
+};
+
+@injectable()
+export class SessionSumaryVariableContribution implements FrontendVariableContribution, AIVariableResolver {
+    protected summaries = new Map<string, { label: string; summary: string; length: number }>();
+    @inject(QuickInputService) protected readonly quickInputService: QuickInputService;
+    @inject(ChatService) protected readonly chatService: ChatService;
+    @inject(ChatSessionSummaryAgent) protected readonly summaryAgent: ChatSessionSummaryAgent;
+
+    registerVariables(service: FrontendVariableService): void {
+        service.registerResolver(SESSION_SUMMARY_VARIABLE, this);
+        service.registerArgumentPicker(SESSION_SUMMARY_VARIABLE, this.pickSession.bind(this));
+        service.registerArgumentCompletionProvider(SESSION_SUMMARY_VARIABLE, this.provideCompletionItems.bind(this));
+    }
+
+    protected async pickSession(): Promise<string | undefined> {
+        const items = this.getItems();
+        const selection = await this.quickInputService.showQuickPick(items);
+        return selection?.id;
+    }
+
+    protected async provideCompletionItems(
+        model: monaco.editor.ITextModel,
+        position: monaco.Position,
+        matchString?: string
+    ): Promise<monaco.languages.CompletionItem[] | undefined> {
+        const context = AIVariableCompletionContext.get(SESSION_SUMMARY_VARIABLE.name, model, position, matchString);
+        if (!context) { return undefined; }
+        const { userInput, range, prefix } = context;
+        return this.getItems().filter(candidate => QuickPickItem.is(candidate) && candidate.label.startsWith(userInput)).map(({ label, id }: QuickPickItem) => ({
+            label,
+            kind: monaco.languages.CompletionItemKind.Class,
+            range,
+            insertText: `${prefix}${id}`,
+            detail: id,
+            filterText: userInput,
+        }));
+    }
+
+    protected getItems(): QuickPickItemOrSeparator[] {
+        return [
+            ...(this.summaries.size ? [{ type: 'separator', label: 'Recent Summaries' }] satisfies QuickPickSeparator[] : []),
```

Comment:
Question 3:
Instead of holding the items here, we discussed to introduce a "Task Context Service" that is responsible for creating and managing task summaries. By default it holds task contexts in memory, but you can set a directory. If a directory is set, the task contexts are persisted as .md files there.
In the IDE package, we introduce a user setting with the default .prompts/task-contexts which sets the directory, so in the IDE tasks are persisted by default. Fine?